packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -80,6 +81,18 @@ class TextPainter {
     _needsLayout = true;
   }
 
+  /// The string used to ellipsize overflowing text.
```

Comment:
Add more documentation, for example:
* Is providing a string sufficient to cause ellipsising?
* Is there a difference between null and the empty string? (If not, consider disallowing one just to remove that axis of freedom.)
* What are the performance implications of this feature?
* What are edge cases you might run into, and how are they handled? e.g. if the ellipsising string is wider than the container, if it contains unbalanced bidi characters, combining characters, or explicit line breaks, that kind of thing.
* How is the ellipsising done? (clipping truncation vs character truncation vs ligature truncation etc)

Weave all these answers into a coherent narrative. :-)